grakn-engine/src/main/java/ai/grakn/engine/controller/api/AttributeTypeController.java

```diff
@@ -0,0 +1,144 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package ai.grakn.engine.controller.api;
+
+import ai.grakn.GraknTx;
+import ai.grakn.GraknTxType;
+import ai.grakn.Keyspace;
+import ai.grakn.concept.AttributeType;
+import ai.grakn.engine.factory.EngineGraknTxFactory;
+import ai.grakn.exception.GraknServerException;
+import mjson.Json;
+import org.apache.commons.httpclient.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.Request;
+import spark.Response;
+import spark.Service;
+
+import java.util.Optional;
+
+import static ai.grakn.engine.controller.util.Requests.mandatoryBody;
+import static ai.grakn.engine.controller.util.Requests.mandatoryPathParameter;
+import static ai.grakn.engine.controller.util.Requests.mandatoryQueryParameter;
+import static ai.grakn.util.REST.Request.ATTRIBUTE_TYPE_LABEL_PARAMETER;
+import static ai.grakn.util.REST.Request.KEYSPACE;
+import static ai.grakn.util.REST.WebPath.Api.ATTRIBUTE_TYPE;
+
+/**
+ * <p>
+ *     AttributeType endpoints
+ * </p>
+ *
+ * @author Ganeshwara Herawan Hananda
+ */
+
+public class AttributeTypeController {
+    private final EngineGraknTxFactory factory;
+    private static final Logger LOG = LoggerFactory.getLogger(AttributeTypeController.class);
+
+    public AttributeTypeController(EngineGraknTxFactory factory, Service spark) {
+        this.factory = factory;
+        spark.post(ATTRIBUTE_TYPE, this::postAttributeType);
+        spark.get(ATTRIBUTE_TYPE + "/" + ATTRIBUTE_TYPE_LABEL_PARAMETER, this::getAttributeType);
+    }
+
+    private Json postAttributeType(Request request, Response response) {
+        LOG.info("postAttributeType - request received.");
+        Json requestBody = Json.read(mandatoryBody(request));
+        String attributeTypeLabel = requestBody.at("attributeType").at("label").asString();
+        String attributeTypeDataTypeRaw = requestBody.at("attributeType").at("type").asString();
+        AttributeType.DataType<?> attributeTypeDataType = fromString(attributeTypeDataTypeRaw);
+        String keyspace = mandatoryQueryParameter(request, KEYSPACE);
+        LOG.info("postAttributeType - attempting to add new attributeType " + attributeTypeLabel + " of type " + attributeTypeDataTypeRaw);
+        try (GraknTx tx = factory.tx(Keyspace.of(keyspace), GraknTxType.WRITE)) {
+            AttributeType attributeType = tx.putAttributeType(attributeTypeLabel, attributeTypeDataType);
+            tx.commit();
+            String jsonConceptId = attributeType.getId().getValue();
```

Comment:
Not sure what happens between a commit and a close but maybe it's better if we do the json conversion outside of the transaction